server/info.go

```diff
@@ -658,6 +661,35 @@ func (s *infoServer) mergeAdditionalInfoToTxs(txs []*types.Transaction, receipts
 		tx.TxFee = txFeeInHydro.String()
 
 		receiptIndex++
+
+		// write external contract data to database
+		if strings.EqualFold(tx.To, cfg.ParamsContractAddr) && tx.Status == 1 && decoded != nil {
+			ctx := context.Background()
+			if decoded.MethodName == "addVote" {
+				proposalID, _ := new(big.Int).SetString(decoded.Arguments["proposalId"].(string), 10)
+				voteOption := new(big.Int).SetInt64(int64(decoded.Arguments["option"].(uint8)))
+				proposal, err := s.kaiClient.GetProposalDetails(ctx, proposalID)
+				if err != nil {
+					s.logger.Info("cannot get proposal by ID using RPC", zap.Any("proposal", proposalID), zap.Error(err))
+					continue
+				}
+				err = s.dbClient.AddVoteToProposal(ctx, proposal, voteOption.Uint64())
+				if err != nil {
+					s.logger.Info("cannot add vote to new proposal in db", zap.Any("decoded", decoded), zap.Error(err))
+				}
+			} else if decoded.MethodName == "confirmProposal" {
+				proposalID, _ := new(big.Int).SetString(decoded.Arguments["proposalId"].(string), 10)
+				proposal, err := s.kaiClient.GetProposalDetails(ctx, proposalID)
+				if err != nil {
+					s.logger.Info("cannot get proposal by ID using RPC", zap.Any("proposal", proposalID), zap.Error(err))
+					continue
+				}
+				err = s.dbClient.UpsertProposal(ctx, proposal)
+				if err != nil {
+					s.logger.Info("cannot confirm proposal in db", zap.Any("decoded", decoded), zap.Error(err))
+				}
+			}
+		}
```

Comment:
Why here ?

Reply:
The transactions which vote and confirm a proposal must be successfully executed (`tx.Status = 1`). I must check in here because there is no event omitted in transaction in case the vote/confirmation is success.